### PR TITLE
Add missing waitForMigrations for workers

### LINKED ISSUE #31624: waitForMigrations.enabled is missing for "workers" helm
### Official Helm Chart version

1.9.0 (latest released)

### Apache Airflow version

2.6.1

### Kubernetes Version

1.26

### Helm Chart configuration

`waitForMigrations` does not have an `enabled` property for the `workers` section. For consistency with the other components and to fully support ignoring migrations, the `enabled` boolean should be added.

### Docker Image customizations

_No response_

### What happened

I tried to set `waitForMigration.enabled` on `workers` but it is not supported.

### What you think should happen instead

For consistency with the other components and to fully support ignoring migrations, the `enabled` boolean should be added.

### How to reproduce

Add this to your `values.yaml`:

```
workers:
  waitForMigrations:
    enabled: false
```

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by grjones**
Add missing `waitForMigrations.enabled` flag and associated logic for `workers` section in helm chart. All other components have this flag. 

closes: #31624

### CODE REVIEW

**Comment by jedcunningham** [comment]
Good find. Can you add a test like [this](https://github.com/apache/airflow/blob/956aac2b2d719e76c9c8cf6a77e9373021900307/tests/charts/airflow_core/test_scheduler.py#L73-L83) in [here](https://github.com/apache/airflow/blob/956aac2b2d719e76c9c8cf6a77e9373021900307/tests/charts/airflow_core/test_worker.py)?

**Comment by grjones** [comment]
> Good find. Can you add a test like [this](https://github.com/apache/airflow/blob/956aac2b2d719e76c9c8cf6a77e9373021900307/tests/charts/airflow_core/test_scheduler.py#L73-L83) in [here](https://github.com/apache/airflow/blob/956aac2b2d719e76c9c8cf6a77e9373021900307/tests/charts/airflow_core/test_worker.py)?

Sure thing. Added. Let me know if there is anything else I can do.

File: tests/charts/airflow_core/test_worker.py [hunk 1]
```diff
@@ -82,6 +82,18 @@ def test_should_add_extra_containers(self):
...
+        docs = render_chart(
+            values={
+                "workers": {
+                    "waitForMigrations": {"enabled": False},
+                },
+            },
+            show_only=["templates/workers/worker-deployment.yaml"],
+        )
+        actual = jmespath.search("spec.template.spec.initContainers", docs[0])
+        assert actual is None
```
Comment (hussein-awala) [inline tests/charts/airflow_core/test_worker.py]: If we add a new init container by default to the worker pod, this test will fail automatically, I wonder if we should assert that the  `initContainers` object is None or it's a list but doesn't contain the `wait-for-airflow-migrations` container. WDYT?

Comment (grjones) [reply tests/charts/airflow_core/test_worker.py]: Yep, I totally agree. I was trying to keep the same pattern as the other `waitforMigrations` tests, but yes, this should be made less brittle.  I'll push up a change later tonight. 👍

Comment (jedcunningham) [reply tests/charts/airflow_core/test_worker.py]: Let's just make this specific one less brittle, then open a separate PR for the others.

Comment (grjones) [reply tests/charts/airflow_core/test_worker.py]: Ok, I pushed up that change to narrow the scope of the test.

**Review by jedcunningham — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

File: tests/charts/airflow_core/test_worker.py [hunk 2]
```diff
@@ -82,6 +82,18 @@ def test_should_add_extra_containers(self):
...
+    def test_disable_wait_for_migration(self):
+        docs = render_chart(
+            values={
+                "workers": {
+                    "waitForMigrations": {"enabled": False},
+                },
+            },
+            show_only=["templates/workers/worker-deployment.yaml"],
+        )
+        actual = jmespath.search("spec.template.spec.initContainers[?name=='wait-for-airflow-migrations']", docs[0])
```
Comment (hussein-awala) [inline tests/charts/airflow_core/test_worker.py]:
```suggestion
        actual = jmespath.search(
            "spec.template.spec.initContainers[?name=='wait-for-airflow-migrations']", docs[0]
        )
```

**Pull request merged**